### PR TITLE
[BOLT] Emit intra-function control flow in YAMLBAT

### PR DESCRIPTION
**Pull request by aaupov**
Attach branch counters to YAML profile, covering intra-function control
flow.

Depends on: https://github.com/llvm/llvm-project/pull/86353

Test Plan: Updated bolt/test/X86/bolt-address-translation-yaml.test

### CODE REVIEW

File: bolt/include/bolt/Profile/BoltAddressTranslation.h [hunk 1]
```diff
@@ -128,6 +128,26 @@ class BoltAddressTranslation {
   /// Returns BB index by function output address (after BOLT) and basic block
   /// input offset.
   unsigned getBBIndex(uint64_t FuncOutputAddress, uint32_t BBInputOffset) const;
+
+  using BBHashMap = std::map<uint32_t, std::pair<unsigned, size_t>>;
+  /// Return a mapping from basic block input offset to hash and block index for a given function.
+  const BBHashMap &getBBHashMap(uint64_t OutputAddress) const {
+    return FuncHashes.at(OutputAddress).second;
```
Comment (rafaelauler) [inline bolt/include/bolt/Profile/BoltAddressTranslation.h:135]: I realized you used "at" a number of times now in this stack of diffs. Isn't this supposed to throw exceptions? Isn't LLVM codebase exception free?

Comment (aaupov) [reply bolt/include/bolt/Profile/BoltAddressTranslation.h:135]: Thanks for pointing out. You're right, it does throw an exception: https://en.cppreference.com/w/cpp/container/map/at

I had code with `find` and `assert` but switched to `at` for conciseness. Let me unwind that.

Comment (aaupov) [reply bolt/include/bolt/Profile/BoltAddressTranslation.h:135]: Addressed in https://github.com/llvm/llvm-project/pull/86353. Will rebase this and follow-up diffs shortly.

**Review by rafaelauler — APPROVED** [review]
LG

**Pull request merged**